src/sidebar/components/test/MarkdownEditor-test.js

```diff
@@ -311,173 +311,52 @@ describe('MarkdownEditor', () => {
-      });
-
-      it('changes focus to the last element when pressing `end`', () => {
-        pressKey('End'); // move to the end node
-        matchesFocusedText('Preview');
-        testRovingIndex();
-      });
-
-      it('changes focus to the first element when pressing `home`', () => {
-        pressKey('ArrowRight'); // move focus off first button
-        pressKey('Home');
-        matchesFocusedTitle('Bold');
-        testRovingIndex();
-      });
+    function testArrowKeySequence(buttons) {
+      for (let button of buttons) {
+        pressKey('ArrowRight');
+        const label =
+          document.activeElement.getAttribute('title') ||
+          document.activeElement.innerText;
+        assert.include(label, button);
+      }
+    }
 
-      it('preserves the elements order and roving index', () => {
-        [
-          {
-            title: 'Italic',
-          },
-          {
-            title: 'Quote',
-          },
-          {
-            title: 'Insert link',
-          },
-          {
-            title: 'Insert image',
-          },
-          {
-            title: 'Insert math (LaTeX is supported)',
-          },
-          {
-            title: 'Numbered list',
-          },
-          {
-            title: 'Bulleted list',
-          },
-          {
-            title: 'Formatting help',
-          },
-          {
-            text: 'Preview',
-          },
-          {
-            // back to the start
-            title: 'Bold',
-          },
-        ].forEach(test => {
-          pressKey('ArrowRight');
-          if (test.title) {
-            matchesFocusedTitle(test.title);
-          }
-          if (test.text) {
-            matchesFocusedText(test.text);
-          }
-          testRovingIndex();
-        });
-      });
+    // This is a basic test that arrow key navigation is enabled.
+    // `useArrowKeyNavigation` tests cover behavior in more detail.
+    it('arrow keys navigate through buttons', () => {
+      // Sequence of buttons we expect to be focused when the first action
+      // ("Bold") action is initially focused and we press the right arrow key
+      // until focus returns to it.
+      const buttons = [
+        'Italic',
+        'Quote',
+        'Insert link',
+        'Insert image',
+        'Insert math',
+        'Numbered list',
+        'Bulleted list',
+        'Formatting help',
+        'Preview',
+        'Bold',
+      ];
+      testArrowKeySequence(buttons);
     });
 
-    context('when `isPreviewing` is true', () => {
-      beforeEach(() => {
-        // turn on Preview mode
-        act(() => {
-          wrapper.find('Toolbar').props().onTogglePreview();
-        });
-        const previewButton = wrapper
-          .find('button')
-          .filterWhere(el => el.text() === 'Write');
-        previewButton.simulate('focus');
-        pressKey('Home');
-      });
+    it('arrow keys navigate through enabled buttons when `isPreviewing` is true', () => {
```

Comment:
This test description is a little garbled grammar-wise.

Reply:
Yeah, fitting a test description into the `it <does X>` grammar doesn't always work. There is a `specify` function which is an alias for `it`, but we don't use that anywhere else in the code.